lxc/list.go

```diff
@@ -281,16 +283,16 @@ func (c *listCmd) listContainers(d *lxd.Client, cinfos []shared.ContainerInfo, f
 	}
 
 	for _, cInfo := range cinfos {
-		cStatesLock.Lock()
-		cSnapshotsLock.Lock()
 		for _, column := range columns {
 			if column.NeedsState && cInfo.IsActive() {
 				_, ok := cStates[cInfo.Name]
 				if ok {
 					continue
 				}
 
+				cStatesLock.Lock()
 				cStates[cInfo.Name] = nil
+				cStatesLock.Unlock()
```

Comment:
They're not quite identical. We won't be holding the lock while adding to the queue anymore, which is apparently enough to fix things here (I do have a system that reproduces the issue).